### PR TITLE
LSR-755 New modular architecture based on an Abstract Factory Pattern

### PR DESCRIPTION
**Pull request by nhpatt**

### CODE REVIEW

File: android/library/core/src/main/java/com/liferay/mobile/screens/base/FactoryProvider.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+import com.liferay.mobile.screens.context.LiferayServerContext;
+
+/**
+ * @author Javier Gamarra
+ */
+public class FactoryProvider {
+
+	public static AbstractFactory getInstance() {
+		if (_abstractFactory == null) {
+			synchronized ("factory") {
```
Comment (jmnavarro) [inline android/library/core/src/main/java/com/liferay/mobile/screens/base/FactoryProvider.java]: Are you sure this synchronization is right? If "factory" is a new String object for every call, then it won't never block

File: android/library/core/src/main/java/com/liferay/mobile/screens/base/FactoryProvider.java [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+		}
+		return _abstractFactory;
+	}
+
+	private static AbstractFactory createFactory() {
+		try {
+			return (AbstractFactory) Class.forName(LiferayServerContext.getFactoryClass()).newInstance();
+		}
+		catch (Exception e) {
+			Log.e("LiferayScreens", "Error creating the instance class");
```
Comment (jmnavarro) [inline android/library/core/src/main/java/com/liferay/mobile/screens/base/FactoryProvider.java]: Add better error message: at least explain that the configuration key may be missing (if so) or the class cannot be found

File: android/library/core/src/main/java/com/liferay/mobile/screens/base/FactoryProvider.java (hunk 1 above)
Comment (nhpatt) [inline android/library/core/src/main/java/com/liferay/mobile/screens/base/FactoryProvider.java]: Yep, that's the equivalent of a static field in the class so just 1 thread could be entering, but it doesn't follow the conventions, changed to FactoryProvider.class.

File: android/library/core/src/main/java/com/liferay/mobile/screens/base/FactoryProvider.java (hunk 2 above)
Comment (nhpatt) [inline android/library/core/src/main/java/com/liferay/mobile/screens/base/FactoryProvider.java]: :+1:

**Pull request merged**